option cc_generic_services = true;

package mir.protobuf;

message ConnectParameters {
  required string application_name = 1;
}

message SurfaceParameters {
  required int32 width = 1;
  required int32 height = 2;
  required int32 pixel_format = 3;
}

message SurfaceId {
  required int32 value = 1;
};

message Surface {
  optional SurfaceId id = 1;
  optional int32 width = 2;
  optional int32 height = 3;
  optional int32 pixel_format = 4;
  optional int32 native_buffer = 5;
  
  optional string error = 127;
}

message Void {
  optional string error = 127;
}

message TestFileDescriptors {
  repeated sint32 fd = 1;
  repeated int32  stuff = 2;

  optional string error = 127;
}

service DisplayServer {
  rpc connect(ConnectParameters) returns (Void);
  rpc disconnect(Void) returns (Void);
  rpc create_surface(SurfaceParameters) returns (Surface);
  rpc release_surface(SurfaceId) returns (Void);
  rpc test_file_descriptors(Void) returns (TestFileDescriptors);
}

